flutter_vlc_player/lib/src/vlc_player_controller.dart

```diff
@@ -0,0 +1,883 @@
+  /// its state before the method was called. (i.e. if this method is called whilst media is playing, once the new
+  /// data source has been loaded, the new stream will begin playing.)
+  /// [dataSource] - the URL of the stream to start playing.
+  /// [dataSourceType] - the source type of media.
+  Future<void> _setStreamUrl(
+    String dataSource, {
+    DataSourceType dataSourceType,
+    String package,
+    bool autoPlay,
+    HwAcc hwAcc,
+  }) async {
+    if (!value.initialized || _isDisposed) {
+      return;
+    }
+    value._initialized = false;
+    await _vlcPlayerPlatform.setStreamUrl(
+      _viewId,
+      uri: dataSource,
+      type: dataSourceType,
+      package: package,
+      hwAcc: hwAcc ?? HwAcc.AUTO,
+      autoPlay: autoPlay ?? true,
+    );
+    value._initialized = true;
+    return;
+  }
+
+  /// Starts playing the video.
+  ///
+  /// This method returns a future that completes as soon as the "play" command
+  /// has been sent to the platform, not when playback itself is totally
+  /// finished.
+  Future<void> play() async {
+    if (!value.initialized || _isDisposed) {
+      return;
+    }
+    await _vlcPlayerPlatform.play(_viewId);
+    // This ensures that the correct playback speed is always applied when
+    // playing back. This is necessary because we do not set playback speed
+    // when paused.
+    await setPlaybackSpeed(value.playbackSpeed);
+  }
+
+  /// Pauses the video.
+  Future<void> pause() async {
+    if (!value.initialized || _isDisposed) {
+      return;
+    }
+    await _vlcPlayerPlatform.pause(_viewId);
+  }
+
+  /// stops the video.
+  Future<void> stop() async {
+    if (!value.initialized || _isDisposed) {
+      return;
+    }
+    await _vlcPlayerPlatform.stop(_viewId);
+  }
+
+  /// Sets whether or not the video should loop after playing once.
+  Future<void> setLooping(bool looping) async {
+    if (!value.initialized || _isDisposed) {
+      return;
+    }
+    value = value.copyWith(isLooping: looping);
+    await _vlcPlayerPlatform.setLooping(_viewId, looping);
+  }
+
+  /// Returns true if media is playing.
+  Future<bool> isPlaying() async {
+    return await _vlcPlayerPlatform.isPlaying(_viewId);
+  }
+
+  /// Set video timestamp in millisecond
+  Future<void> setTime(int time) async {
+    return await seekTo(Duration(milliseconds: time));
+  }
+
+  /// Sets the video's current timestamp to be at [moment]. The next
+  /// time the video is played it will resume from the given [moment].
+  ///
+  /// If [moment] is outside of the video's full range it will be automatically
+  /// and silently clamped.
+  Future<void> seekTo(Duration position) async {
+    if (!value.initialized || _isDisposed) {
+      return null;
+    }
+    if (position > value.duration) {
+      position = value.duration;
+    } else if (position < const Duration()) {
+      position = const Duration();
+    }
```

Comment:
Nit: we could be a bit more explicit here using `Duration.zero`
```suggestion
    } else if (position < Duration.zero) {
      position = Duration.zero;
    }
```

Reply:
fixed.